ext/opentelemetry-ext-flask/tests/conftest.py

```diff
@@ -0,0 +1,24 @@
+# Copyright 2020, OpenTelemetry Authors
```

Comment:
@c24t I think you had a strong opinion on avoiding pytest magic. Is this OK for you?

Reply:
When the test module `test_flask_integration.py` is executed this happens:

```python
...
from flask import Flask, request
...
```
If we don't do the patching before that line gets executed, then the original, unpatched `Flask` class gets imported. Even if we patch afterwards, the unpatched class remains, so when we get to run this:

```python

def setUp(self):
    ...
    self.app = Flask(__name__)
```
`self.app` will be an instance of the unpatched `Flask` class. That is why I am using `pytest_sessionstart` and `pytest_sessionfinish` hooks here, because they get executed before Pytest does its test collection. This behavior is more closely related to the actual behavior that comes with using the command line patcher who does is patching before the interpreter starts via `sitecustomize.py`. It is a good point that this makes the patching and unpatching happen only once per session and not once per test case and that could be necessary, even if it is not necessary right now for the tests to pass. Another solution could be to put the import of `Flask` in `setUp` after a call to `FlaskPatcher().patch()`. I think the disadvantage of this approach is that it forces the test writer to know about the tiny details that surround this issue by carefully keeping the import of `Flask` after the patching is done.

Reply:
Sorry to leave this comment hanging! I've pushed back on other pytest magic -- especially autouse fixtures -- when they make the tests hard to reason about, modify global state, or do some surprising thing like start a docker instance.

This looks to me like a good use of sessionstart/finsh to me because it keeps the tests from worrying about finicky import-order effects, or whether the integration is loaded.